genesyscloud/resource_genesyscloud_telephony_didpool.go

```diff
@@ -0,0 +1,202 @@
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validatePhoneNumber,
+			},
+			"end_phone_number": {
+				Description:      "Ending phone number of the DID Pool range.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validatePhoneNumber,
+			},
+			"description": {
+				Description: "DID Pool description.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"comments": {
+				Description: "Comments for the DID Pool.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"pool_provider": {
+				Description:  "Provider (PURE_CLOUD | PURE_CLOUD_VOICE).",
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"PURE_CLOUD", "PURE_CLOUD_VOICE"}, false),
+			},
+		},
+	}
+}
+
+func createDidPool(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	startPhoneNumber := d.Get("start_phone_number").(string)
+	endPhoneNumber := d.Get("end_phone_number").(string)
+	description := d.Get("description").(string)
+	comments := d.Get("comments").(string)
+	poolProvider := d.Get("pool_provider").(string)
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	telephonyApi := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
+
+	log.Printf("Creating DID pool %s", startPhoneNumber)
+	didPool, _, err := telephonyApi.PostTelephonyProvidersEdgesDidpools(platformclientv2.Didpool{
+		StartPhoneNumber: &startPhoneNumber,
+		EndPhoneNumber:   &endPhoneNumber,
+		Description:      &description,
+		Comments:         &comments,
+		Provider:         &poolProvider,
+	})
+	if err != nil {
+		return diag.Errorf("Failed to create DID pool %s: %s", startPhoneNumber, err)
+	}
+
+	d.SetId(*didPool.Id)
+
+	log.Printf("Created DID pool %s %s", startPhoneNumber, *didPool.Id)
+	return readDidPool(ctx, d, meta)
+}
+
+func readDidPool(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	telephonyApi := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
+
+	log.Printf("Reading DID pool %s", d.Id())
+	didPool, resp, getErr := telephonyApi.GetTelephonyProvidersEdgesDidpool(d.Id())
+	if getErr != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Failed to read DID pool %s: %s", d.Id(), getErr)
+	}
+
+	if *didPool.State == "deleted" {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("start_phone_number", *didPool.StartPhoneNumber)
+	d.Set("end_phone_number", *didPool.EndPhoneNumber)
+
+	if didPool.Description != nil {
+		d.Set("description", *didPool.Description)
+	} else {
+		d.Set("description", nil)
+	}
+
+	if didPool.Comments != nil {
+		d.Set("comments", *didPool.Comments)
+	} else {
+		d.Set("comments", nil)
+	}
+
+	if didPool.Provider != nil {
+		d.Set("pool_provider", *didPool.Provider)
```

Comment:
Formatting looks off on this.  Make sure you run go fmt on the file.  Go normally ident so many spaces over.

Reply:
I ran gofmt for the files but didn't really notice any difference in indentation.